test/plain-list/AutosuggestApp.test.js

```diff
@@ -592,6 +615,42 @@ describe('Default Autosuggest', () => {
       focusAndSetInputValue(' e');
       expectSuggestions([]);
     });
+
+    describe("when ignoring certain reasons", () => {
+      beforeEach(() => {
+        shouldRenderSuggestions.callsFake((value, reason) => {
+          return reason !== 'input-focused'; // Show suggestions always, except on input focus
+        });
+      });
+
+      afterEach(() => {
+        shouldRenderSuggestions.resetBehavior();
+        shouldRenderSuggestions.callsFake(defaultShouldRenderSuggestionsStub);
+      });
+
+      it('should be able to show suggestions only in certain cases', () => {
+        focusInput();
+        expectSuggestions([]);
+
+        clickUp();
+        expectSuggestions([
```

Comment:
So here we show all possible items, even though the input field is empty, because of `getMatchingLanguages`'s regexp?

Reply:
Yes, exactly, because `new RegExp("^", 'i').test("anything")` is always true, and that's what we effectively end up doing with an empty input (and in reality, my application does something pretty similar too).

Reply:
Ok! Good. Just something one might not expect from that function name. But then, all good :)